samples/snippets/src/main/java/com/example/firestore/snippets/MultipleRangeInequalityFiltersSnippets.java

```diff
@@ -0,0 +1,150 @@
+    sfData.put("name", "San Francisco");
+    sfData.put("state", "CA");
+    sfData.put("country", "USA");
+    sfData.put("capital", false);
+    sfData.put("population", 860000L);
+    sfData.put("density", 18000L);
+    sfData.put("regions", Arrays.asList("west_coast", "norcal"));
+
+    Map<String, Object> laData = new HashMap<>();
+    laData.put("name", "Los Angeles");
+    laData.put("state", "CA");
+    laData.put("country", "USA");
+    laData.put("capital", false);
+    laData.put("population", 3900000L);
+    laData.put("density", 8300L);
+    laData.put("regions", Arrays.asList("west_coast", "socal"));
+
+    Map<String, Object> dcData = new HashMap<>();
+    dcData.put("name", "Washington D.C.");
+    dcData.put("state", null);
+    dcData.put("country", "USA");
+    dcData.put("capital", true);
+    dcData.put("population", 680000L);
+    dcData.put("density", 11300L);
+    dcData.put("regions", Arrays.asList("east_coast"));
+
+    Map<String, Object> tokData = new HashMap<>();
+    tokData.put("name", "Tokyo");
+    tokData.put("state", null);
+    tokData.put("country", "Japan");
+    tokData.put("capital", true);
+    tokData.put("population", 9000000L);
+    tokData.put("density", 16000L);
+    tokData.put("regions", Arrays.asList("kanto", "honshu"));
+
+    Map<String, Object> bjData = new HashMap<>();
+    bjData.put("name", "Beijing");
+    bjData.put("state", null);
+    bjData.put("country", "China");
+    bjData.put("capital", true);
+    bjData.put("population", 21500000L);
+    bjData.put("density", 3500L);
+    bjData.put("regions", Arrays.asList("jingjinji", "hebei"));
+
+    CollectionReference citiesCollection = db.collection("cities");
+
+    // Add each city document
+    addCityDocument(citiesCollection, "SF", sfData);
+    addCityDocument(citiesCollection, "LA", laData);
+    addCityDocument(citiesCollection, "DC", dcData);
+    addCityDocument(citiesCollection, "TOK", tokData);
+    addCityDocument(citiesCollection, "BJ", bjData);
+
+    System.out.println("Data added successfully!");
+  }
+
+  private static void addCityDocument(CollectionReference citiesRef, String documentId,
+      Map<String, Object> data) {
+    ApiFuture<WriteResult> future = citiesRef.document(documentId).set(data);
+    try {
+      WriteResult result = future.get();
+      System.out.println("Update time : " + result.getUpdateTime());
+    } catch (Exception e) {
+      System.err.println("Error adding document: " + e.getMessage());
+    }
+  }
+
+  /* Example of Query with range and inequality filters. */
+  Query compoundMultiInequalities() {
+    // [START firestore_query_filter_compound_multi_ineq]
+    Query query = db.collection("cities")
+        .whereGreaterThan("population", 1000000)
+        .whereLessThan("density", 10000);
+    // [END firestore_query_filter_compound_multi_ineq]
+    return query;
+  }
+
+  /* Example of filtering indexes considering performance reasons. */
+  void indexingConsiderations() {
```

Comment:
Is this being used anywhere?

Reply:
I didn't create a unit test for it because we don't have any of the sample data available for it. I was just going to paste in the region tag.